src/global/core/components/animations/animation.ts

```diff
@@ -0,0 +1,12 @@
+import { AbstractAnimation } from './animation-abstract';
+import { AnimationWeb } from './animation-web';
+import { AnimationCss } from './animation-css';
+
+const supportsWebAnimations = (): boolean => (typeof (AnimationEffect as any) === 'function' || typeof (window as any).AnimationEffect === 'function') &&
+  (typeof (Element as any) === 'function') && (typeof (Element as any).prototype.animate === 'function');
+
+export const createAnimation = (animationId?: string): AbstractAnimation => (
+  supportsWebAnimations()
+    ? new AnimationWeb(animationId)
+    : new AnimationCss(animationId)
```

Comment:
I think we only need to check this once. For this to change, a different execution context would be required in which case the code would be executed once more anyway.
```suggestion
const supportsWebAnimations: boolean = (typeof (AnimationEffect as any) === 'function' || typeof (window as any).AnimationEffect === 'function') &&
  (typeof (Element as any) === 'function') && (typeof (Element as any).prototype.animate === 'function');
const AnimationVariant = supportsWebAnimations ? AnimationWeb : AnimationCss;

export const createAnimation = (animationId?: string): AbstractAnimation => AnimationVariant(animationId);
```